In [1]:
import numpy as np
import pandas as pd
from pathlib import Path

In [2]:
train_df = pd.read_csv(Path('/content/2019loans.csv'))
test_df = pd.read_csv(Path('/content/2020Q1loans.csv'))

In [3]:
# Convert categorical data to numeric and separate target feature for training data
X_train=pd.get_dummies(train_df.drop(columns=["target"]))
Y_train_df=train_df["target"]

In [4]:
# Convert categorical data to numeric and separate target feature for testing data
X_test=pd.get_dummies(test_df.drop(columns=["target"]))
Y_test_df=test_df["target"]



In [5]:
# add missing dummy variables to testing set
for column in X_train.columns:
  if column not in X_test.columns:
    X_test[column]=0

In [7]:
# Train the Logistic Regression model on the unscaled data and print the model score
from sklearn.linear_model import LogisticRegression
model=LogisticRegression()
model.fit(X_train,Y_train_df)
model.score(X_test, Y_test_df)

/usr/local/lib/python3.7/dist-packages/sklearn/linear_model/_logistic.py:818: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG,


0.5082943428328371

In [8]:
# Train a Random Forest Classifier model and print the model score
from sklearn.ensemble import RandomForestClassifier
model_2=RandomForestClassifier()
model_2.fit(X_train,Y_train_df)
model_2.score(X_test, Y_test_df)

0.6380263717566993

In [9]:
# Scale the data
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
scaler.fit(X_train)
X_trainScaled= scaler.transform(X_train)
X_testScaled= scaler.transform(X_test)

In [10]:
# Train the Logistic Regression model on the scaled data and print the model score
Model_3=LogisticRegression()
Model_3.fit (X_trainScaled,Y_train_df)
Model_3.score(X_testScaled,Y_test_df)


/usr/local/lib/python3.7/dist-packages/sklearn/linear_model/_logistic.py:818: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG,


0.7598894087622289

In [11]:
# Train a Random Forest Classifier model on the scaled data and print the model score
Model_4= RandomForestClassifier()
Model_4.fit(X_trainScaled, Y_train_df)
Model_4.score(X_testScaled, Y_test_df)


0.6365376435559337

The best model turned out to be a scaled logisticregression model witht a accurracy of about 76%